src/app/settings/components/project-role-edit-panel/project-role-jwt-tokens.tsx

```diff
@@ -0,0 +1,143 @@
+
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import { FormApi, Text } from 'react-form';
+
+import { FormField } from '../../../shared/components';
+import { AppContext } from '../../../shared/context';
+import { JwtToken } from '../../../shared/models';
+import { CreateJWTTokenParams, DeleteJWTTokenParams, services } from '../../../shared/services';
+
+interface ProjectRoleJWTTokensProps {
+    tokens: JwtToken[];
+    api: FormApi;
```

Comment:
No need to reuse form from parent component. Define own form in ProjectRoleJWTTokens and use form `onSubmit` to trigger token creation.

Reply:
Okay I separated it into a separate field.